src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.GetCommand;
+import duke.command.ListCommand;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Makes sense of the user command.
+ */
+public class Parser {
+    /**
+     * Parses the specified command and returns the corresponding Command object.
+     *
+     * @param command The command to be parsed.
+     * @return The corresponding Command object.
+     * @throws DukeException If the command is not supported.
+     */
+    public static Command parse(String command) throws DukeException {
+        Command cmd;
+        String[] commandArr = command.split(" ", 2);
+        switch (commandArr[0]) {
+        case "todo":
+            try {
+                cmd = new AddCommand(commandArr[0], commandArr[1]);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            }
+            break;
+        case "deadline":
+            try {
+                cmd = new AddCommand(commandArr[0], commandArr[1]);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("The description and due date of a deadline cannot be empty.");
+            }
+            break;
+        case "event":
+            try {
+                cmd = new AddCommand(commandArr[0], commandArr[1]);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("The description and date and time of an event cannot be empty.");
+            }
+            break;
+        case "done":
+            try {
+                if (commandArr[1].trim().equals("")) {
```

Comment:
Nice use of string.trim()! I had some trouble with leading and trailing spaces. Will incorporate into my Duke HAHA. Thanks!!